### DISCUSSION TITLE
Current state of proprietary vs open source nvidia drivers

### DISCUSSION THREAD

**Discussion by Blayung** [NVIDIA/open-gpu-kernel-modules#457] [Q&A] [answered]
Once nvidia made these open source drivers, people said that it's not a good idea to switch to the open source drivers yet, because it's still much slower than the proprietary ones and that we need to wait a little bit until the community makes it actually worth using them. Is that still the case?

**Comment by aritger** [comment, marked as answer]
Whether open-gpu-kernel-modules are currently sufficient for you depends on your use case.  E.g., most compute workloads perform comparably between open-gpu-kernel-modules and the proprietary kernel modules.  Graphics on open-gpu-kernel-modules have a few performance issues we're actively working through, and there are some missing features we're working to address (most notably power management, but also things like variable refresh rate).

I would suggest that you test it out for yourself to decide if the current state of open-gpu-kernel-modules is good for your uses.

Either way, we're working to close any gaps, so that eventually there will be no reason to not use open-gpu-kernel-modules.

**Comment by gentuser** [comment]
Hi everybody,
Do you still find this to be the case after three months of active development?

**Comment by 5had3z** [comment]
Based on the language of the latest cuda version download [page](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=22.04&target_type=deb_local), "legacy kernel module" vs "open kernel module". Is it now recommended to use the open kernel module given the option?

**Comment by izmyname** [comment]
RTD3 isn't supported yet.

**Comment by Sturmlocke86** [comment]
What is the current state on the open source nvidia drivers for all things desktop (Gnome, Wayland, X11), screen recording via OBS (nvenc) and gaming through Steam's Proton and all things Flatpak? I don't mind if power management is still wip and the card runs at max power the entire time, that's how I use it on my desktop anyway. Can anyone report back their feedback; how well do the open source drivers perform now? Ty

**Comment by Anyeos** [comment]
@Sturmlocke86 I am asking the same question.

**Comment by dolphin-cat** [comment]
how's the nvidia opensource driver doing now in 2024? is it usable for a desktop? does it have any major disadvantages when compared to the proprietary or is it the same?

**Reply by craschnet** [reply]
From the developer forums:
https://forums.developer.nvidia.com/t/clarifying-560-series-drivers-open-sourcedness-vs-kernel-module-type-proprietary/292698

> In general, by release 560, the open kernel modules should have approximate feature parity with the proprietary kernel modules. I would categorize things into several buckets:

Also you'll want to read:
https://forums.developer.nvidia.com/t/unix-graphics-feature-deprecation-schedule/60588

> Starting in the release 560 series, it will be recommended to use the [open flavor of NVIDIA Linux Kernel Modules 507] (http://us.download.nvidia.com/XFree86/Linux-x86_64/550.78/README/kernel_open.html) wherever possible (Turing or later GPUs, or Ada or later when using GPU virtualization).

Since I have an A2 card for testing and docker, I'm sticking with proprietary. Time to upgrade to an L4 or a 4000 Ada

**Reply by tuxayo** [reply]
@dolphin-cat For performances, see https://www.phoronix.com/review/nvidia-r550-open

**Comment by Al-faqun** [comment]
Hi there from early 2026. I'm now choosing between using open or closed kernel drivers. Has open module achieved full parity for features and performance, compared to legacy drivers? I've struggled to find comprehensive info on this topic. Phoronix article, linked above, suggests that in 2024 there were still some issues:

> But there are some known limitations in the open kernel driver with not yet having G-SYNC support on notebooks and vRAM is not preserved across power management yet.

I've also seen tickets about performance issues related to GSP on KDE.

Are they fixed? Is there any reason to stick to closed kernel module on Ampere?

**Reply by ptr1337** [reply]
Hi,

The GSP Issue on KDE is fixed and not present anymore since 575.

The only issue which I would know of is D3Cold on Turing not properly working with the open modules, otherwise the open modules are the suggested way to use NVIDIA on Linux :)

**Reply by Al-faqun** [reply]
Thank you for a swift reply, I am gonna use open kernel modules then.

**Comment by rvdende** [comment]
No question the open is better just simply because if you run into an issue it is fixable.

Edit: unless there is some issue that needs access to undocumented gpu registers... 

Nvidia please just open the proprietary drivers the community can take it further then. Saves you dev costs, community would love it and we all get better value long term
